File: command/java/Waiter.java
package command.java;

public class Waiter {
    private Action action1;
    private Action action2;

    public Waiter(Action action1, Action action2) {
        this.action1 = action1;
        this.action2 = action2;
    }

    public void executeActions() {
        if (action1 != null)
            action1.Doit();
        if (action2 != null)
            action2.Doit();
    }
}

File: command/golang/waiter.go
package main

// Waiter acts as the invoker
type Waiter struct {
	actions []Action
}

func (w *Waiter) AddAction(a Action) {
	w.actions = append(w.actions, a)
}

func (w *Waiter) ExecuteActions() {
	for _, action := range w.actions {
		action.DoIt()
	}
	// Clear actions after execution
	w.actions = nil
}
